components/style/values/specified/length.rs

```diff
@@ -458,6 +458,17 @@ impl Parse for Length {
     }
 }
 
+impl Either<Length, None_> {
+    #[inline]
+    #[allow(missing_docs)]
+    pub fn parse_non_negative_length(context: &ParserContext, input: &mut Parser) -> Result<Self, ()> {
+        if input.try(|input| None_::parse(context, input)).is_ok() {
+            return Ok(Either::Second(None_));
+        }
+        Length::parse_internal(input, AllowedNumericType::NonNegative).map(Either::First)
```

Comment:
No, it's just that we already have the same impl for `LengthOrNormal` and `LengthOrAuto`, and now we have one more for `LengthOrNone` (which is literally the same). It's better if we clean it up when we're touching that code. Also, note that these are all `<value-keyword>` kinda things (so, only one will be numeric at any time).